src/user-lists/user-list-events-stats.service.ts

```diff
@@ -443,6 +444,238 @@ export class UserListEventsStatsService {
     return entities;
   }
 
+  async findContributorCategoriesByTimeframe(
+    options: ContributionsTimeframeDto,
+    listId: string
+  ): Promise<DbContributorCategoryTimeframe[]> {
+    const range = options.range!;
+
+    const allUsers = await this.findContributorsByType(listId, range, UserListContributorStatsTypeEnum.all);
+
+    if (allUsers.length === 0) {
+      return new Array<DbContributorCategoryTimeframe>();
```

Comment:
nit: return the terse form of an empty array. No need to type it as it will be inferred by the return type of the function.

```suggestion
      return [];
```

Reply:
👍🏼 I was doing this in a few places so mutated all those. Thanks for the tip!!